storm-core/src/jvm/backtype/storm/spout/RawScheme.java

```diff
@@ -18,11 +18,13 @@
 package backtype.storm.spout;
 
 import backtype.storm.tuple.Fields;
+
+import java.nio.ByteBuffer;
 import java.util.List;
 import static backtype.storm.utils.Utils.tuple;
 
 public class RawScheme implements Scheme {
-    public List<Object> deserialize(byte[] ser) {
+    public List<Object> deserialize(ByteBuffer ser) {
```

Comment:
@harshach Actually I was referring to the receiver (the bolts) that might be currently doing something like `byte[] bytes = inputTuple.getBinaryByField("bytes");` to get the data emitted from kafka spout. It appears that the `deserialize` method returns a tuple that wraps the input ByteBuffer which gets emitted.